tc/core/polyhedral/memory_promotion.cc

```diff
@@ -328,17 +328,38 @@ void addSingletonReferenceGroups(
 }
 } // namespace
 
-TensorGroups TensorReferenceGroup::accessedBySubtree(
-    const ScheduleTree* tree,
-    const Scop& scop) {
+// Compute tensor reference groups encapsulating all tensor accesses within
+// "outerSchedule".  Only statement instances present in the domain of
+// "outerSchedule" are considered.  In particular, if this domain is
+// intersected with block and/or thread mapping, the reference groups are
```

Comment:
It seems to me that the comment about groups that are "private" to a mapping really come as a byproduct of the IR. Generic parameters are understood to take one value in the parameter domain whereas mapping parameters are understood to take all values in the set but are represented as generic parameters. So the consequence for this function (and other functions that don't treat mapping paremeters in a special way is that you get the "private" behavior. Does that sound like a reasonable description? Worth to mention in the comment?

Reply:
> That means that at the point where you introduced
> parameters for block/thread identifiers, you are
> considering one particular block/thread.

This is exactly how I see it, and that's why that function can compute groups specific to blocks and/or threads.  It _is_ considering one particular block/thread, it just can express everything given the id of that block thread so it returns the result in a compact form.  It is the same as parametric domains and what @nicolasvasilache calls generic parameters.  Parameters will have one fixed value, but the schedule is computed so that it remains valid for all possible values of those parameters.

It took me some time to wrap my head around it in context of the code generation, but it is actually easier to reason this way.